libimage/push.go

```diff
@@ -86,8 +86,8 @@ func (r *Runtime) Push(ctx context.Context, source, destination string, options
 		destRef = dockerRef
 	}
 
-	// docker-archive and only DockerV2Schema2MediaType support Gzip compression
-	if options.CompressionFormat != nil &&
+	// docker-archive and DockerV2Schema2MediaType support only Gzip compression
+	if options.CompressionFormat != nil && options.CompressionFormat.Name() != "gzip" &&
```

Comment:
`c/image/pkg/compression/types.GzipAlgorithmName`, please.

---

(Absolutely non-blocking: The way I think about things, it seems more natural to me to order the conditions `(compressionFormat != nil && (/* supports gzip only */) && /* compressionFormat is not gzip */)`, but that’s a personal aesthetic preference, and there are good arguments for both the current ordering and one other ordering.)

Reply:
For me, it would be weird to separate the nil check from the type check - the nil check is really just a guard and we only really need to care if the compression format isn't gzip.

The order also helps minimise the amount of work to answer the if statement.

So, I would leave the ordering alone. LMK if you disagree as really, I also don't mind switching it.